pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/multithreading/UnsynchronizedStaticFormatterRule.java

```diff
@@ -69,7 +69,7 @@ public Object visit(ASTFieldDeclaration node, Object data) {
             }
 
             ASTMethodDeclaration method = n.getFirstParentOfType(ASTMethodDeclaration.class);
```

Comment:
With my comments on the test cases, we need to look out for a ASTSynchronizedStatement here instead of the method...
Simple case would be, if there is no synchronized statement found, then we add a violation.

We could however also check, that the synchronized statement actually synchronizes over the variable - we know the name of the variable, it is `var.getVariableName()`. If we have the synchronized statement, we could check it like this:

```java
ASTSynchronizedStatement syncStatement = n.getFirstParentOfType(ASTSynchronizedStatement.class);
if (syncStatement != null) {
    ASTExpression expression = syncStatement.getFirstChildOfType(ASTExpression.class);
    if (expression != null) {
        ASTName name = expression.getFirstDescendantOfType(ASTName.class);
        if (name != null && name.hasImageEqualTo(var.getVariableName())) {
            continue;
        }
    }
}
```

Reply:
@adangel 
Thanks for your comments.
I experimented something like this for synchronized blocks.
However as I mentioned in the PR, We would not be able to determine the scope of the locking since it can be an **expression.**

For example consider below.

private static final SimpleDateFormat sdf = new SimpleDateFormat();
    void bar() {
        synchronized (getFormat()) {
            sdf.format("s");
        }
    }
    
    static SimpleDateFormat getFormat() {
        return sdf;
    }

Reply:
IMHO, I would also raise a violation in this rule, if the synchronized expression is not the field. The rule starts by looking at the fields of type "*Format". And if you have such a field, I don't see a reason, why to use a method to get the field...

`synchronized (getFormat())` could be valid, if the formatter is defined elsewhere (e.g. a subclass), e.g.

```java
void bar() {
    SimpleDateFormat sdf = getFormat();
    synchronized (sdf) {
        sdf.format("s");
    }
}

protected abstract SimpleDateFormat getFormat();
```

Note: you'll need to use a temporary local variable, since you might get different result objects back from getFormat() - and in this case, you would synchronize on a different object than you'd use later on....